Tests/SentryTests/SentryOptionsTest.m

```diff
@@ -17,28 +17,37 @@ @interface SentryOptionsTest : XCTestCase
 
 @implementation SentryOptionsTest
 
-// TODO(fetzig): not sure if this test needs an update or SentryOptions/SentryDsn needs a fix.
-//- (void)testEmptyDsn {
-//    NSError *error = nil;
-//    SentryOptions *options = [[SentryOptions alloc] initWithDict:@{} didFailWithError:&error];
-//    XCTAssertEqual(kSentryErrorInvalidDsnError, error.code);
-//    XCTAssertNil(options);
-//}
+
+- (void)testEmptyDsn {
+    NSError *error = nil;
+    SentryOptions *options = [[SentryOptions alloc] initWithDict:@{} didFailWithError:&error];
+    
+    // TODO(fetzig): not sure if this test needs an update or SentryOptions/SentryDsn needs a fix. Maybe the error should be set to kSentryErrorInvalidDsnError
+    [self assertDidabled:options andError:error];
+}
+
+- (void)testInvalidDsnBoolean {
+    NSError *error = nil;
+    SentryOptions *options = [[SentryOptions alloc] initWithDict:@{@"dsn": @YES} didFailWithError:&error];
+    
+    // TODO(fetzig): not sure if this test needs an update or SentryOptions/SentryDsn needs a fix. Maybe the error should be set to kSentryErrorInvalidDsnError
+    [self assertDidabled:options andError:error];
+}
+
+-(void)assertDidabled: (SentryOptions*) options
+               andError: (NSError*) error {
+    XCTAssertNil(options.dsn);
+    XCTAssertEqual(@NO, options.enabled);
```

Comment:
I *think* something like this could work

```suggestion
    XCTAssertFalse(options.enabled);
```

Reply:
Sadly this doesn't work.  [XCTAssertFalse](https://developer.apple.com/documentation/xctest/xctassertfalse) succeeds when `expression == false` and not `expression == @NO`